bushes_classic/init.lua

```diff
@@ -30,27 +30,54 @@ local modpath = minetest.get_modpath('bushes_classic')
 dofile(modpath..'/cooking.lua')
 dofile(modpath..'/nodes.lua')
 
-biome_lib.register_active_spawner({
-	spawn_delay = 3600,
-	spawn_plants = bushes_classic.spawn_list,
-	avoid_radius = 10,
-	spawn_chance = 100,
-	spawn_surfaces = {
+local spawn_plants = bushes_classic.spawn_list
+
+local function get_biome_data(pos, perlin_fertile)
+	local fertility = perlin_fertile:get_2d({x=pos.x, y=pos.z})
+
+	local data = minetest.get_biome_data(pos)
+	-- Original values this method returned were +1 (lowest) to -1 (highest)
+	-- so we need to convert the 0-100 range from get_biome_data() to that.
+	return fertility, 1 - (data.heat / 100 * 2), 1 - (data.humidity / 100 * 2)
+end
+
+minetest.register_abm({
+	nodenames = {
 		"default:dirt_with_grass",
 		"woodsoils:dirt_with_leaves_1",
 		"woodsoils:grass_with_leaves_1",
 		"woodsoils:grass_with_leaves_2",
 		"farming:soil",
 		"farming:soil_wet"
 	},
-	avoid_nodes = {"group:bush"},
-	seed_diff = 545342534, -- chosen by a fair mashing of the keyboard - guaranteed to be random :P
-	plantlife_limit = -0.1,
-	light_min = 10,
-	temp_min = 0.15, -- approx 20C
-	temp_max = -0.15, -- approx 35C
-	humidity_min = 0, -- 50% RH
-	humidity_max = -1, -- 100% RH
+	interval = 3600,
+	chance = 100,
+	label = "[bushes_classic] spawn bushes",
+	min_y = -16,
+    max_y = 48,
+	action = function(pos, node)
+		local p_top = {x = pos.x, y = pos.y + 1, z = pos.z}
+		local n_top = minetest.get_node_or_nil(p_top)
+		if not n_top then return end
+		if n_top.name ~= "air" then return end
```

Comment:
unless line lengths prohibit it, combine this to one line?
`if not n_top or n_top.name~="air" then return end`